src/components/Header.tsx

```diff
@@ -42,33 +35,17 @@ export interface IHeader {
 }
 
 export function Header(props: IHeader) {
-  const [isModalVisible, setIsModalVisible] = useState(false);
-  const { abortSession, recordingStatus, togglePause, toggleRecording } =
-    useContext(RecordingContext);
+  const { urlRef } = useContext(UrlContext);
+  const { recordingStatus, toggleRecording } = useContext(RecordingContext);
```

Comment:
The main impetus for having the Pause functionality is to allow them to click on things in the browser without PW recording it. This could be to close a popover element or something of the like that blocks the view of the main DOM elements, which wouldn't be a problem for Synthetics to click behind, but would add unnecessary complexity to the script if the close procedure were recorded. There are probably other cases for this as well that @vigneshshanmugam could detail.

I do think that it could be good if Pause were a less prominent function, and Stop took its place though.

> shouldn't we make it a boolean isRecording?

Until very recently this is what we had. But we also had an `isPaused` flag, and the code to reason out the actual state between two boolean flags was more convoluted than using an enum to cover the three states, which is what we've done. If we eliminated the Pause feature then we'd revert back to using a boolean. 👍

Reply:
The pause functionality was inherited from the PW recorder where it was useful to pause the recording to debug/develop or do some changes in the UI. 

It does increase the complexity, I agree.